src/viam/resource/manager.py

```diff
@@ -99,13 +101,27 @@ async def remove_resource(self, name: ResourceName):
         """
         with self._lock:
             try:
-                resource = self._resource_by_name_only(name.name)
+                resource = self.resources[name]
                 await resource.close()
             except Exception as e:
                 raise e
             finally:
                 del self.resources[name]
 
+    async def close(self):
+        """Close the resourcce manager by removing all resources"""
+        rns = [key for key in self.resources.keys()]
+        with self._lock:
+            errors = {}
+            for rn in rns:
+                try:
+                    await self.remove_resource(rn)
+                except Exception as e:
+                    errors[rn] = e
+                    LOGGER.error(f"Error while closing {rn.name}:", e)
+            if errors != {}:
+                raise Exception(errors)
```

Comment:
I think it's ok to not raise even with errors, would just note in the docstring.

Reply:
Sounds good. Thank you!